src/models/StyledComponent.js

```diff
@@ -58,11 +58,16 @@ export default (ComponentStyle: any) => {
         }
       }
 
+      determineExecutionContext(props: any, theme: any) {
+        if (!theme) return Object.assign({}, this.props)
+        return Object.assign({}, this.props, { theme })
+      }
+
       /* eslint-disable react/prop-types */
       render() {
         const { className, children } = this.props
-        const theme = this.state.theme || {}
-        const executionContext = Object.assign({}, this.props, { theme })
+        const theme = this.state.theme
```

Comment:
This is what I get what I get for coding so late. This was actually what I attempted first and it wasn't fixing my test case. I'm guessing I didn't have the build/test suite running correctly at that point, sorry. Will update the PR with the much simpler version.